### PR TITLE
Improve Cancelable and Network cancellation ❌

### PR DESCRIPTION
**Pull request by p4checo**
# Changes

- Add `isCancelable` property to `Cancelable`.

- Improve `CancelableBag` to cancel any added cancelables if already cancelled, to ensure correctness and avoid lingering work that can't be cancelled.

- Added more checks in `URLSessionNetworkStack` to avoid scheduling retry work if the cancelable has already been cancelled.

- Handle `NSURLErrorCancelled` error in `NetworkStack`'s `NetworkStore` default conformance.

- Create extension on `NSError` to check if it's a `NSURLErrorDomain` and `NSURLErrorCancelled` error.

- Cleaned up Info.plist placement and added missing one for `HostAppRequiringTests` target.

### CODE REVIEW

File: Sources/Stores/NetworkPersistableStore.swift [hunk 1]
```diff
@@ -242,8 +242,7 @@ where Network.Remote == Data, Network.Request == URLRequest, Network.Response ==
             switch result {
             case .success(let data):
                 success(data)
-            case .failure(.url(let error as NSError))
-            where error.domain == NSURLErrorDomain && error.code == NSURLErrorCancelled:
+            case .failure(.url(let error as NSError)) where error.isURLErrorCancelled:
```
Comment (jwelton) [inline Sources/Stores/NetworkPersistableStore.swift]: Just out of curiosity. Couldn't you switch on the error type instead? I.e.
``` swift
case .failure(.url(URLError.cancelled))
```

Comment (p4checo) [reply Sources/Stores/NetworkPersistableStore.swift]: Great point! Even though we can (and probably should) use `URLError`, I think that we cannot switch on `URLError.cancelled`, as that's only the code of the error 😓. Still, I think we can go with:

`case .failure(.url(let error as URLError)) where error.code == .cancelled:`

😉

File: Tests/AlicerceTests/Shared/CancelableBagTestCase.swift [hunk 2]
```diff
@@ -0,0 +1,117 @@
...
+
+    func testIsCancelled_WithCancelledCancelableBag_ShouldReturnTrue() {
+        XCTAssertFalse(cancelable.isCancelled)
+        cancelable.cancel()
+        XCTAssertTrue(cancelable.isCancelled)
+    }
+
+    // add
+
+    func testAdd_WithNotCancelledCancelledBag_ShouldAddCancelableToBag() {
```
Comment (filipe-lemos) [inline Tests/AlicerceTests/Shared/CancelableBagTestCase.swift]: Maybe use `WithNotCancelledCancelableBag` or simply `WithNotCancelledBag` as the `Bag`s in the test refer to `CancelableBag` anyway.

Comment (p4checo) [reply Tests/AlicerceTests/Shared/CancelableBagTestCase.swift]: I think I've messed up those names, yes 😆

**Pull request merged**